### PR TITLE
feat: added support for maintenance version

### LINKED ISSUE #610: Terrform Postgres Module creates MySQL Instance - Clarify documentation
### TL;DR

Using the Postgres module creates a MySQL Instance using the same values provided in the README.md

### Expected behavior

It should create a POSTGRES_9_6 database instance 

### Observed behavior

It creates MySQL 8.0 version Instance 

### Terraform Configuration

```hcl
module "pg" {
  # checkov:skip=CKV_GCP_62:
  # checkov:skip=CKV_TF_1:
  source                          = "terraform-google-modules/sql-db/google//modules/postgresql"
  version                         = "~> 20.1"
  name                            = var.instance_name
  random_instance_name            = true
  project_id                      = var.project_id
  database_version                = var.db_version
  region                          = var.region
  db_name                         = var.db_name
  disk_size                       = var.db_disk_size
  disk_type                       = var.db_disk_type
  db_charset                      = "UTF8"
  db_collation                    = "en_US.UTF8"
  user_name                       = "${var.prefix}_user"
  user_password                   = var.password
  maintenance_window_day          = 7
  maintenance_window_hour         = 12
  maintenance_window_update_track = "stable"

  database_flags = [
    { name = "log_connections", value = "on" },
    { name = "log_min_duration_statement", value = "-1" },
    { name = "log_disconnections", value = "on" },
    { name = "log_lock_waits", value = "on" },

  ]

  ip_configuration = {
    ipv4_enabled       = false
    require_ssl        = false
    private_network    = var.vpc_network
    allocated_ip_range = null
    # authorized_networks = [
    #   {
    #     name  = "${var.project_id}-cidr"
    #     value = var.pg_ha_external_ip_range
    #   },
    # ]
  }

  backup_configuration = {
    enabled                        = true
    start_time                     = "20:55"
    location                       = null
    point_in_time_recovery_enabled = false
    transaction_log_retention_days = null
    retained_backups               = 90
    retention_unit                 = "COUNT"
  }
}
```

### Terraform Version

```sh
required_version = ">= 0.13"
```

### Additional information

_No response_

### LINKED ISSUE #662: Missing support for maintenance_version
### TL;DR

The option maintenance_version is missing in the Terraform SQL module. It should be supported for managing SQL database maintenance versions, but currently, this feature does not exist.

### Expected behavior

The maintenance_version option should be a supported parameter within the Terraform SQL module, allowing users to specify or track the version of a database specifically for maintenance purposes.

### Observed behavior

_No response_

### Terraform Configuration

```hcl
https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/sql_database_instance#maintenance_version
```

### Terraform Version

```sh
1.9.8
```

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by imrannayer**
fixed #610 
fixed #662

### CODE REVIEW

**Comment by imrannayer** [comment]
@q2w plz review

**Review by g-awmalik — APPROVED** [review]
LGTM but can you check if this requires a version bump for the modules?

**Comment by imrannayer** [comment]
Min version 4.39s required for it. We are at 6.1
